### PR TITLE
fix(decorator): add character limit, display value in `title` prop

### LINKED ISSUE #591: DataDecorator for long text behaviour changed
Expected behavior -

When there is a long text inside a data decorator, on V1 it used to replace it with elipsis. 

Actual behavior -

Long text is shown as is no matter how long
image

Steps for reproducing
Check out the storybook & add very long text:
https://ibm-security.carbondesignsystem.com/?path=/story/components-decorator--default

### PR DESCRIPTION
**Pull request by SimonFinney**
## Affected issue

Resolves #591

## Proposed changes

- Set 30-character limit for the `Decorator` value
- Update `Decorator` to display `value` if no `title` is provided
- Use Carbon's `text-overflow` helper method
- Share `height` and `line-height` value in case this changes

### CODE REVIEW

**Comment by SimonFinney** [comment]
@petervachon Based on some of the conversations in the originating issue's thread, I've added a 30-character limit for truncating the value of the `Decorator` - you can test this out by using the knobs in the deploy preview above!

Let me know how this feels, and if we need to make any adjustments - thanks!

File: src/components/DataDecorator/Decorator/Decorator.js [hunk 1]
```diff
@@ -37,25 +37,29 @@ class Decorator extends Component {
...
-        <span className={`${namespace}__icon`}>
-          <Icon
-            fillRule="evenodd"
-            path={path}
-            size={inline ? 12 : 16}
-            viewBox="0 0 16 16"
-            title={this.props.scoreDescription(score, scoreThresholds)}
-          />
+  renderDecorator = (noIcon, path, inline, score, scoreThresholds) => {
+    const { scoreDescription, title, type, value } = this.props;
```
Comment (jendowns) [inline src/components/DataDecorator/Decorator/Decorator.js:41]: There's a lot going of destructuring happening here and around line 66 (with some of those destructured props from line 66 being passed to this `renderDecorator`). If this PR is going to include some cleanup, can it be done to include the destructure around line 66?

File: src/components/DataDecorator/_mixins.scss [hunk 2]
```diff
@@ -29,6 +29,12 @@
 @mixin decorator {
   $root: &;
 
+  /// Height.
+  /// @type Length
+  $data-decorator_sizing__height: carbon--mini-units(
```
Comment (jendowns) [inline src/components/DataDecorator/_mixins.scss:34]:
```suggestion
  $data-decorator__sizing__height: carbon--mini-units(
```

File: src/components/DataDecorator/_mixins.scss [hunk 3]
```diff
@@ -37,7 +43,7 @@
   display: inline-flex;
   box-sizing: border-box;
   font-family: carbon--font-family($name: sans);
-  height: carbon--mini-units(3.25);
+  height: $data-decorator_sizing__height;
```
Comment (jendowns) [inline src/components/DataDecorator/_mixins.scss:46]:
```suggestion
  height: $data-decorator__sizing__height;
```

File: src/components/DataDecorator/_mixins.scss [hunk 4]
```diff
@@ -178,10 +184,8 @@
...
 
   &__value {
-    text-overflow: ellipsis;
-    white-space: nowrap;
-    overflow: hidden;
-    display: inline;
-    line-height: 1.625rem;
+    @include text-overflow;
+    max-width: 30ch;
+    line-height: $data-decorator_sizing__height;
```
Comment (jendowns) [inline src/components/DataDecorator/_mixins.scss:189]:
```suggestion
    line-height: $data-decorator__sizing__height;
```

File: src/components/DataDecorator/Decorator/Decorator.js (hunk 1 above)
Comment (SimonFinney) [reply src/components/DataDecorator/Decorator/Decorator.js:41]: Good catch!

**Comment by SimonFinney** [comment]
@petervachon Are you referring to the 'Update `Decorator` to display `value` if no `title` is provided' change? Currently, the `Decorator` already allows support for a `title` tooltip attribute as per a previous change - we're now applying the `value` to the tooltip attribute if a consumer has not explicitly provided a `title`.

**Comment by petervachon** [comment]
@SimonFinney yes I was referring to 'Update Decorator to display value if no title is provided'. I didn't realize you were using it to provide the tooltip attribute, I thought it was referring to the value in the actual decorator component, so please disregard. Thanks!

**Pull request merged**